### ISSUE TITLE
Change dotnet new to create portable pdbs by default

### ISSUE THREAD

**Issue by sayedihashimi** [dotnet/cli#3421]
The current experience for VS Code users is not ideal for debugging when running on Mac/Linux because you cannot step into a break point until you update `project.json` to produce portable pdbs by default.

We would like to default this to create portable pdbs when running `dotnet build`, but after some discussions we feel that is too risky for 1.0 RTM. Instead for 1.0 RTM we would like to update templates in `dotnet new` and yeoman. For 1.1 I will create a new issue to change the default in `dotnet build`

Here is what we need to add.

```
  "buildOptions": {
    "debugType": "portable"
  },
```

**Comment by Petermarcu** [comment]
Moved the milestone. Will @ajaybhargavb be making the PR?

**Comment by ajaybhargavb** [comment]
Yes. I am waiting for https://github.com/dotnet/cli/pull/3275 and https://github.com/dotnet/cli/pull/2106 to go in so I can update all of them

**Comment by eerhardt** [comment]
Why not take this change instead?  https://github.com/dotnet/cli/pull/2356.  If there is no "debugType" specified - then "portable" is the default.

**Comment by ajaybhargavb** [comment]
@Petermarcu @eerhardt, Are we still doing this?

**Comment by piotrpMSFT** [comment]
Fixed by #3468

**Comment by kuncevic** [comment]
In my case I generated aspnet webapi project using https://github.com/omnisharp/generator-aspnet#readme and didn't get debugging work then I tried just to generate pure .net console app using `dotnet new`. I find out that debugger just worked for console app. So then I found there is a difference in `buildOptions` section of `project.json` files. The webapi `project.json` missed `"debugType": "portable"`. After I add that the debug just working fine for me. Also I noticed the `.dll` output has different size - that is proving point that with `"debugType": "portable"` flag compiler producing different assembly.

**Comment by kuncevic** [comment]
In my case I generated aspnet webapi project using https://github.com/omnisharp/generator-aspnet#readme and didn't get debugging work then I tried just to generate pure .net console app using `dotnet new`. I find out that debugger just worked for console app. So then I found there is a difference in `buildOptions` section of `project.json` files. The webapi `project.json` missed `"debugType": "portable"`. After I add that the debug just working fine for me. Also I noticed the `.dll` output has different size - that is proving point that with `"debugType": "portable"` flag compiler producing different assembly.
